native/src/main/java/io/ballerina/lib/sap/BallerinaDestinationDataProvider.java

```diff
@@ -51,11 +51,15 @@ public boolean supportsEvents() {
 
     public void addDestination(BMap<BString, Object> jcoDestinationConfig, BString destinationName) {
         Properties properties = new Properties();
-        jcoDestinationConfig.entrySet().forEach(entry -> {
-            BString key = entry.getKey();
-            BString value = (BString) entry.getValue();
-            properties.setProperty(SAPConstants.CONFIG_KEYS.get(key.toString()), value.toString());
-        });
-        destinationProperties.put(destinationName.toString(), properties);
+        try {
+            jcoDestinationConfig.entrySet().forEach(entry -> {
+                BString key = entry.getKey();
+                BString value = (BString) entry.getValue();
+                properties.setProperty(SAPConstants.CONFIG_KEYS.get(key.toString()), value.toString());
+            });
+            destinationProperties.put(destinationName.toString(), properties);
+        } catch (Exception e) {
+            throw new RuntimeException("Error while adding destination: " + e.getMessage());
```

Comment:
```suggestion
            throw new RuntimeException("Error while processing destination properties: " + e.getMessage());
```

Reply:
Add e also to the error

Reply:
Changed with [010b6b2](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/12/commits/010b6b2c237f70b1e7db75e29c8c780db22968de).